python/lsst/ts/atmonochromator/model.py

```diff
@@ -68,6 +68,8 @@ async def disconnect(self) -> None:
             try:
                 writer.write_eof()
                 await asyncio.wait_for(writer.drain(), timeout=2)
+            except Exception:
+                self.log.exception("Failed to disconnect")
             finally:
                 writer.close()
```

Comment:
I wonder it this line will fail in the case where `writer.write_eof()` fails. I guess we will find out.

Reply:
I suppose setting the writer to none might ensure it is closed?